prefect_aws/lambda_function.py

```diff
@@ -0,0 +1,195 @@
+    ).invoke(tail=True)
+    ```
+
+    Invoke a lambda function with a client context
+
+    ```python
+    LambdaFunction(
+        function_name="test-function",
+        aws_credentials=aws_credentials,
+    ).invoke(client_context={"bar": "foo"})
+    ```
+
+"""
+import json
+from typing import Any, Literal, Optional, Tuple
+
+import boto3
+from prefect.blocks.core import Block
+from prefect.utilities.asyncutils import run_sync_in_worker_thread, sync_compatible
+from pydantic import VERSION as PYDANTIC_VERSION
+
+if PYDANTIC_VERSION.startswith("2."):
+    from pydantic.v1 import Field
+else:
+    from pydantic import Field
+
+from prefect_aws.credentials import AwsCredentials
+
+
+class LambdaFunction(Block):
+    """Invoke a Lambda function. This block is part of the prefect-aws
+    collection. Install prefect-aws with `pip install prefect-aws` to use this
+    block.
+
+    Attributes:
+        function_name: The name, ARN, or partial ARN of the Lambda function to
+            run. This must be the name of a function that is already deployed
+            to AWS Lambda.
+        qualifier: The version or alias of the Lambda function to use when
+            invoked. If not specified, the latest (unqualified) version of the
+            Lambda function will be used.
+        aws_credentials: The AWS credentials to use to connect to AWS Lambda
+            with a default factory of AwsCredentials.
+
+    """
+
+    _block_type_name = "Lambda Function"
+    _logo_url = "https://cdn.sanity.io/images/3ugk85nk/production/d74b16fe84ce626345adf235a47008fea2869a60-225x225.png"  # noqa
+    _documentation_url = "https://prefecthq.github.io/prefect-aws/s3/#prefect_aws.lambda_function.LambdaFunction"  # noqa
+
+    function_name: str = Field(
+        title="Function Name",
+        description=(
+            "The name, ARN, or partial ARN of the Lambda function to run. This"
+            " must be the name of a function that is already deployed to AWS"
+            " Lambda."
+        ),
+    )
+    qualifier: Optional[str] = Field(
+        default=None,
+        title="Qualifier",
+        description=(
+            "The version or alias of the Lambda function to use when invoked. "
+            "If not specified, the latest (unqualified) version of the Lambda "
+            "function will be used."
+        ),
+    )
+    aws_credentials: AwsCredentials = Field(
+        title="AWS Credentials",
+        default_factory=AwsCredentials,
+        description="The AWS credentials to invoke the Lambda with.",
+    )
+
+    class Config:
+        """Lambda's pydantic configuration."""
+
+        smart_union = True
+
+    def _get_session_and_client(self) -> Tuple[boto3.Session, Any]:
+        """
+        Retrieve a boto3 session and Lambda client
+        """
+        boto_session = self.aws_credentials.get_boto3_session()
+        lambda_client = boto_session.client("lambda")
+        return boto_session, lambda_client
```

Comment:
Looks like the session isn't used, so I think it'd be worthwhile to simplify this function.
```suggestion
    def _get_lambda_client(self):
        """
        Retrieve a boto3 session and Lambda client
        """
        boto_session = self.aws_credentials.get_boto3_session()
        lambda_client = boto_session.client("lambda")
        return lambda_client
```